proxmoxtf/resource/vm.go

```diff
@@ -3807,24 +3827,8 @@ func vmReadCustom(
 		diskMap[di] = disk
 	}
 
-	var keyList []string
-
-	for key := range diskMap {
-		keyList = append(keyList, key)
-	}
-
-	sort.Strings(keyList)
-
-	for _, k := range keyList {
-		orderedDiskList = append(orderedDiskList, diskMap[k])
-	}
-
-	if len(clone) > 0 {
-		if len(currentDiskList) > 0 {
-			err := d.Set(mkResourceVirtualEnvironmentVMDisk, orderedDiskList)
-			diags = append(diags, diag.FromErr(err)...)
-		}
-	} else if len(currentDiskList) > 0 {
```

Comment:
It feels like something's missing from the original condition,... but your change makes sense -- the both branches are essentially the same.

Reply:
Yeah, there's definitely something wrong with disk management when the VM comes from a clone. I intend to take a look at it at some point, mostly because of this bug which is kind of annoying.